languages/zCEE3.groovy

```diff
@@ -0,0 +1,96 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import groovy.transform.*
+import com.ibm.dbb.build.report.*
+import com.ibm.dbb.build.report.records.*
+import java.nio.file.*;
+
+
+// define script properties
+@Field BuildProperties props = BuildProperties.getInstance()
+@Field def buildUtils= loadScript(new File("${props.zAppBuildDir}/utilities/BuildUtilities.groovy"))
+	
+println("** Building ${argMap.buildList.size()} ${argMap.buildList.size() == 1 ? 'file' : 'files'} mapped to ${this.class.getName()}.groovy script")
+
+// verify required build properties
+buildUtils.assertBuildProperties(props.zCEE_requiredBuildProperties)
+
+// sort the build list based on build file rank if provided
+List<String> sortedList = buildUtils.sortBuildList(argMap.buildList.sort(), 'zCEE_fileBuildRank')
+int currentBuildFileNumber = 1
+
+// iterate through build list
+sortedList.each { buildFile ->
+	println "*** (${currentBuildFileNumber++}/${sortedList.size()}) Building file $buildFile"
+	
+	boolean alwaysRebuildWAR = props.getFileProperty('zCEE_alwaysRebuildWAR', buildFile).toBoolean()
+	
+	String workingDir = buildFile.replace("/src/main/api/openapi.yaml", "")
+	String gradleBuildLocation = "build"
+	if (props.verbose) 
+		println("** Path to build.gradle: ${workingDir}/${gradleBuildLocation}")
+	String WarLocation = "${workingDir}/build/libs/api.war"
+	if (props.verbose) 
+		println("** Expected path to api.war: ${WarLocation}")
+	File WarFile = new File(WarLocation)
+	String[] command;
+	String commandString;
+	
+	if (alwaysRebuildWAR || !WarFile.exists()) {
+		println("** File api.war doesn't exist for this z/OS Connect project. Building it...")
+		String JAVA_OPTS = props.getFileProperty('zCEE_gradle_JAVA_OPTS', buildFile)
+		String gradlePath = props.getFileProperty('zCEE_gradlePath', buildFile)
+		String shellEnvironment = props.getFileProperty('zCEE_shellEnvironment', buildFile)
+	
+		command = [shellEnvironment, gradlePath, gradleBuildLocation]
+		commandString = command.join(" ") 
+		if (props.verbose)
+			println("** Command to execute: ${commandString} - in working directory: ${workingDir}")
+		StringBuffer shellOutput = new StringBuffer()
+		StringBuffer shellError = new StringBuffer()
+	
+		ProcessBuilder cmd = new ProcessBuilder(shellEnvironment, gradlePath, gradleBuildLocation);
+		Map<String, String> env = cmd.environment();
+		env.put("JAVA_OPTS", JAVA_OPTS);
+		cmd.directory(new File(workingDir));
+		Process process = cmd.start()
+		process.consumeProcessOutput(shellOutput, shellError)
+		process.waitFor()
```

Comment:
Can we capture the output of the gradle build in a log file similar to the CICS Resource Builder script, and make sure we store it in the output directory?

https://github.com/IBM/dbb-zappbuild/blob/1996c0436b72787c9e9257defc91fbfc0c7e6add/languages/CRB.groovy#L75-L79

Reply:
Done